R/board_gcs.R

```diff
@@ -0,0 +1,262 @@
+#' Use a Google Cloud Storage bucket as a board
+#'
+#' Pin data to a Google Cloud Storage bucket using the googleCloudStorageR
+#' package.
+#'
+#' # Authentication
+#'
+#' `board_gcs()` is powered by the googleCloudStorageR package which provides
+#' several authentication options, as documented in its
+#' [main vignette](https://code.markedmondson.me/googleCloudStorageR/articles/googleCloudStorageR.html).
+#' The two main options are to create a service account key (a JSON file) or an
+#' authentication token; you can manage either using the [gargle](https://gargle.r-lib.org/) package.
+#'
+#' # Details
+#'
+#' * The functions in pins do not create a new bucket. You can create
+#'   a new bucket from R with [googleCloudStorageR::gcs_create_bucket()].
+#' * You can pass arguments for [googleCloudStorageR::gcs_upload] such as
+#'   `predefinedAcl` and `upload_type` through the dots of `pin_write()`.
+#' * `board_gcs()` is powered by the googleCloudStorageR package, which is a
+#'   suggested dependency of pins (not required for pins in general). If
+#'   you run into errors when deploying content to a server like
+#'   <https://shinyapps.io> or [Connect](https://posit.co/products/enterprise/connect/),
+#'   add `library(googleCloudStorageR)` to your app or document for [automatic
+#'   dependency discovery](https://support.posit.co/hc/en-us/articles/229998627-Why-does-my-app-work-locally-but-not-on-my-RStudio-Connect-server).
+#'
+#' @inheritParams new_board
+#' @param bucket Bucket name. You can only write to an existing bucket. Defaults
```

Comment:
It seems weird to give this a default to me. Not sure if I can articulate why.

Reply:
Yeah, none of our other boards uses this kind of default. I saw that googleCloudStorageR has this support for setting up a default bucket to use via environment variable, and I thought I would integrate it in case people are already doing this. If you feel strongly, I am fine with removing it.